readline/line.go

```diff
@@ -183,26 +184,47 @@ func (rl *Instance) deleteToEnd() {
 	rl.line = rl.line[:rl.pos]
 }
 
-func (rl *Instance) emacsForwardWord(tokeniser tokeniser) (adjust int) {
-	// when emacs has more specific stuff, move this in a file with then
-	split, index, pos := tokeniser(rl.line, rl.pos)
-	if len(split) == 0 {
-		return
+func isAlNum(r rune) (b bool) {
+	return unicode.IsDigit(r) || unicode.IsLetter(r)
+}
+
+func (rl *Instance) emacsForwardWord() (adjust int) {
+	pos := rl.pos
+	end := len(rl.line)
+
+	if pos >= end {
+		return 0
 	}
 
-	word := rTrimWhiteSpace(split[index])
+	// goto first nonalphabet char if not already on one
+	if isAlNum(rl.line[pos]) == false {
+		for pos++; pos < end && !isAlNum(rl.line[pos]); pos++ { }
+	}
 
-	switch {
-	case len(split) == 0:
-		return
-	case index == len(split)-1 && pos >= len(word)-1:
-		return
-	case pos >= len(word)-1:
-		word = rTrimWhiteSpace(split[index+1])
-		adjust = len(split[index]) - pos
-		adjust += len(word)
-	default:
-		adjust = len(word) - pos
+	if pos >= end {
+		return 0
 	}
-	return
+
+	// goto nonalphabet
+	for pos++ ;pos < end && isAlNum(rl.line[pos]); pos++ { }
```

Comment:
the empty for loops are really triggering, could the tokenise function(s) not be used

Reply:
for something that looks better, you can use strings.FieldsFunc and then get len of the first entry

Reply:
IDKK I am dumb. With this I just do directly what emacsForwardWord is supposed to do (go forwards until word). I can try to use tokeniser and prob should I guess. Words are configurable with tokeniser too. What do you use to debug as right now I am just using fmt.Println but obviously thats not very good

Reply:
i just debug print to stderr

Reply:
OKAY I GOT IT WORKING WITH tokenizer PROPERLY. emacsforwardword is kinda scuffed because of the way that the tokenizer deals with whitespace but it works fine without crashing